### PR TITLE
Refactor typescript compile in plugin

### LINKED ISSUE #18: New way to rebuild Anuglar2 model generated by Rhetos Plugin
Right now, when run ```DeployPackage.exe```, the ```Angular2GeneratorPlugin``` will generate **Rhetos.Angular2.ts**, after that it will:
- Copy **Rhetos.Angular2.ts** to ```AdminGuiCompile`` folder (in RhetosBuild) to build it by ```tsc``` for get newest js file. (admingui.js) (1)
- Copy **admingui.js** to ```AdminGui``` folder (in RhetosBuild) to run in Rhetos

But I found an error when run tsc at (1), it use global tsc and throw errors althought **admingui.js** still generated (in some computers).

My way to fix it like this:
- After generating **Rhetos.Angular2.ts**, we will store it somewhere in RhetosBuild. 
- Use build script to copy it to AdminGui Front-End folder and build it to get newest js file. (admingui.js)
- Copy **admingui.js** to ```AdminGui``` folder (in RhetosBuild).

What do you guys think? Is it ok?

### PR DESCRIPTION
**Pull request by huynguyen-95**
This will resolve issue fixes #18 
- Remove unnecessary packages in package.json
- Remove globally **tsc**
- Use local **tsc** and add exception when compiling failed.

### CODE REVIEW

**Review by nguyenlamlll — APPROVED** [review]
So far the build is all good and everything is still normal. The changes look good to me. Do you have anything else, @ngoquoc?

File: AdminGUI/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
         "angular2-in-memory-web-api": "0.0.7",
         "basecode": "1.2.0",
         "es6-shim": "0.35.0",
+        "gulp-typescript": "^4.0.2",
```
Comment (ngoquoc) [inline AdminGUI/package.json]: Should this be moved to `devDependencies` instead?

File: RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs [hunk 2]
```diff
@@ -69,18 +72,36 @@ public void Generate()
...
+            
             process.Start();
-            process.WaitForExit();
+            try
+            {
+                _deployPackagesLogger.Trace(process.StandardOutput);
+                if (process.StandardError.Peek() > -1)
+                {
+                    _deployPackagesLogger.Error(process.StandardError);
+                    throw new TypeScriptException("Compiled TypeScript model file failed.");
```
Comment (ngoquoc) [inline RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs]: Message should be "Compiling TypeScript models failed", or "Failed to compile TypeScript models."

File: AdminGUI/package.json (hunk 1 above)
Comment (huynguyen-95) [reply AdminGUI/package.json]: Agree :+1:

File: RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs (hunk 2 above)
Comment (huynguyen-95) [reply RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs]: Agree :+1:

File: AdminGUI/package.json (hunk 1 above)
Comment (huynguyen-95) [reply AdminGUI/package.json]: Removed gulp-typescript because we won't use it anymore.

**Comment by huynguyen-95** [comment]
@ngoquoc I updated as your comments.

**Pull request merged**